src/server/core.rs

```diff
@@ -156,13 +178,34 @@ impl Core {
 
     pub fn add_task(&mut self, task_ref: TaskRef) {
         let task_id = task_ref.get().id();
-        if task_ref.get().is_ready() {
-            self.ready_to_assign.push(task_ref.clone());
+        {
+            let task = task_ref.get();
+            if task.is_ready() {
+                self.add_ready_to_assign(task_ref.clone());
+            }
         }
         self.has_new_tasks = true;
         assert!(self.tasks.insert(task_id, task_ref).is_none());
     }
 
+    #[inline(never)]
```

Comment:
Why the inline annotations here and below?

Reply:
Because try_wakeup_parked_resources are called often (for every task at least once) but waking almost never happen.

Reply:
Ok. I'm not sure if it's worth it to inline a nontrivial function with a loop, but the compiler can decide anyway.

Reply:
It is more or less forgotten here as there were originally more complex cold/hot paths here, so probably it should be removed now.